src/target/source/codegen_opencl.cc

```diff
@@ -541,12 +541,26 @@ void CodeGenOpenCL::VisitExpr_(const OrNode* op, std::ostream& os) {
 }
 
 void CodeGenOpenCL::VisitExpr_(const SelectNode* op, std::ostream& os) {
+  std::ostringstream oss;
   os << "select(";
-  PrintExpr(op->false_value, os);
+  PrintExpr(op->false_value, oss);
```

Comment:
I'm not sure, but should we also modify `test_opencl_type_casting` after these changes?

Reply:
I think your test case should still be right. And I may add other cases later.

Reply:
I think your test case should still be right. And I may add other cases later.

Reply:
I think it would be nice to have a test which will cover new functionality with adding type casting into select operator.